### PR TITLE
Add proto define of default editor in org-level and repo-level

### PR DESCRIPTION
**Pull request by mustard-mh**
## Description

Add missing property for default editor option

## Related Issue(s)

Relates EXP-1259 project

## How to test

As long as build passed

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/public-api/gitpod/v1/configuration.proto [hunk 1]
```diff
@@ -109,11 +110,15 @@ message UpdateConfigurationRequest {
     optional bool update_restricted_workspace_classes = 3;
 
     // restricted_editor_names specifies the editor names that are NOT allowed to be used in this configuration.
+    // Only applies to UX, the backend does not enforce this.
```
Comment (filiptronicek) [inline components/public-api/gitpod/v1/configuration.proto:113]: nit: I think it should be "UI" here

File: components/public-api/gitpod/v1/configuration.proto [hunk 2]
```diff
@@ -109,11 +110,15 @@ message UpdateConfigurationRequest {
...
     // If empty, all editors are allowed.
     // Only updates if update_restricted_editor_names is true.
     repeated string restricted_editor_names = 4;
+
     // Specifies whether restricted_editor_names should be updated.
     optional bool update_restricted_editor_names = 5;
+
+    optional string default_editor_name = 6;
```
Comment (filiptronicek) [inline components/public-api/gitpod/v1/configuration.proto:121]: nit: perhaps also add a comment like we do for the organization-scoped?
```suggestion
	// Specifies the default editor name for workspaces in the repository.
    optional string default_editor_name = 6;
```

File: components/public-api/gitpod/v1/organization.proto [hunk 3]
```diff
@@ -124,13 +125,17 @@ message UpdateOrganizationSettingsRequest {
   // Pass an empty array to allow all workspace classes
   repeated string allowed_workspace_classes = 5;
 
-  // restricted_editor_names updates the list of restricted editor names that are not allowed to be used by workspaces in an organization.
+  // restricted_editor_names updates the list of restricted editor names that are NOT allowed to be used by workspaces in an organization.
+  // Only applies to UX, the backend does not enforce this.
```
Comment (filiptronicek) [inline components/public-api/gitpod/v1/organization.proto:129]: nit: same as above

**Review by filiptronicek — APPROVED** [review]
LGTM :shipit:

**Comment by mustard-mh** [comment]
Close as we don't need default editor anymore [[internal chat](https://www.notion.so/Project-Spec-Available-Editors-4dcc9795626f4e5587350a7f842bd2ab?d=882ff739ec0846f9afe44e2f5ae14c70&pvs=4#49fa58646fce4448a4f8f84e09a86042)]

**Pull request closed without merging**